### PR TITLE
[CELEBORN-1632] Support to apply ratis local raft_meta_conf command with RESTful api

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?
Sub-task of CELEBORN-1628.

Support to apply ratis local raft_meta_conf with RESTful api.

See https://celeborn.apache.org/docs/latest/celeborn_ratis_shell/#local-raftmetaconf
```
$ celeborn-ratis sh local raftMetaConf -peers <[P0_ID|]P0_HOST:P0_PORT,[P1_ID|]P1_HOST:P1_PORT,[P2_ID|]P2_HOST:P2_PORT> -path <PARENT_PATH_OF_RAFT_META_CONF>
```

The implementation is same with https://github.com/apache/ratis/blob/e96ed1a33840385446f4e647864a169467da5ab7/ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/local/RaftMetaConfCommand.java#L122-L133

### Why are the changes needed?

We have implemented the RESTful implementation for all the others ratis-shell command.

image

| Ratis Shell               | RESTful api                        |
|----------------------|---------------------------------|
| election transfer    | `/ratis/election/transfer`      |
| election stepDown    | `/ratis/election/step_down`     |
| election pause       | `/ratis/election/pause`         |
| election resume      | `/ratis/election/resume`        |
| group info           | `/masters`                      |
| peer add             | `/ratis/peer/add`               |
| peer remove          | `/ratis/peer/remove`            |
| peer setPriority     | `/ratis/peer/set_priority`      |
| snapshot create      | `/ratis/snapshot/create`        |

And the local raftMetaConf command is the last one.

I closed the ticket CELEBORN-1632 before, I thought it is a local command and wonder whether it is necessary to implement it with RESTful api.

But we have implemented all the others, so I decide to implement it as well.

### Does this PR introduce _any_ user-facing change?

A new API.

The implementation is same with https://github.com/apache/ratis/blob/e96ed1a33840385446f4e647864a169467da5ab7/ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/local/RaftMetaConfCommand.java#L122-L133

### How was this patch tested?
image
image
image

image
image

### CODE REVIEW

**Review by SteNicholas — APPROVED** [review]

**Comment by turboFei** [comment]
will self review the logical again, we may need to allow to replace the existing peer.

**Comment by turboFei** [comment]
gentle ping @SteNicholas again

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/http/api/v1/RatisResource.scala [hunk 1]
```diff
@@ -235,6 +245,50 @@ class RatisResource extends ApiRequestContext with Logging {
...
+  @ApiResponse(
+    responseCode = "200",
+    content = Array(new Content(
+      mediaType = MediaType.APPLICATION_OCTET_STREAM,
+      schema = new Schema(implementation = classOf[Response]))),
+    description = "Generate a new-raft-meta.conf file based on original raft-meta.conf" +
+      " and new peers, which is used to move a raft node to a new node.")
+  @POST
+  @Path("/local/raft_meta_conf")
+  @Produces(Array(MediaType.APPLICATION_OCTET_STREAM))
```
Comment (RexXiong) [inline master/src/main/scala/org/apache/celeborn/service/deploy/master/http/api/v1/RatisResource.scala:257]: IMO we need add another method for getting the raw local raft meta conf

**Comment by turboFei** [comment]
image

**Review by RexXiong — APPROVED** [review]

**Comment by RexXiong** [comment]
Thanks, merge to main(v0.6.0)

**Pull request closed without merging**